cdap-explore/src/main/java/co/cask/cdap/explore/service/hive/Hive14ExploreService.java

```diff
@@ -86,7 +89,35 @@ public Hive14ExploreService(TransactionSystemClient txClient, DatasetFramework d
   @Override
   protected QueryStatus doFetchStatus(OperationHandle operationHandle)
     throws HiveSQLException, ExploreException, HandleNotFoundException {
-    OperationStatus operationStatus = getCliService().getOperationStatus(operationHandle);
+    CLIService cliService = getCliService();
+    OperationStatus operationStatus;
+
+    /*
+    Hive 1.2.1000.2.6.0.0-598, that ships with HDP 2.6.0 introduced a change as part of
+    https://issues.apache.org/jira/browse/HIVE-15473 in the getOperationStatus method and introduced a boolean argument
+    to get the progress update for the operation. Previous versions of the method just took an {@link OperationHandle}
+    as the single argument.
+    To handle this we are using reflection to find out how many arguments the method takes and then call it accordingly.
+     */
+    int i;
+    // Find the getOperationStatus method so that we can check the number of arguments it expects.
+    Method[] methods = cliService.getClass().getMethods();
+    for (i = 0; i < methods.length; i++) {
```

Comment:
A little more concise and safe:
```
    OperationStatus operationStatus;
    Method getOperationStatus = null;
    for (Method method : getCliService().getClass().getMethods()) {
      if ("getOperationStatus".equals(method.getName())) {
        getOperationStatus = method;
      }
    }
    if (getOperationStatus == null) {
      // throw exception 
    }
```
Also, you can move this part to the start of the service so it only needs to be done once.